include/gatekeeper_sol.h

```diff
@@ -175,4 +175,10 @@ sol_conf_hold(struct sol_config *sol_conf)
 
 int sol_conf_put(struct sol_config *sol_conf);
 
+static inline void
+set_prio(struct rte_mbuf *pkt, uint8_t priority)
+{
+	((struct sol_mbuf_priv *)rte_mbuf_to_priv(pkt))->priority = priority;
+}
+
```

Comment:
Please add the following function before `set_prio()` and replace **all** occurrences of `((struct sol_mbuf_priv *)rte_mbuf_to_priv(pkt))` in this pull request with `mbuf_to_sol_priv(pkt)`:
```c
static inline struct sol_mbuf_priv *
mbuf_to_sol_priv(struct rte_mbuf *pkt)
{
	return rte_mbuf_to_priv(pkt);
}
```
This helps with type safety and centralizes the use of `rte_mbuf_to_priv()`.